### PR TITLE
feat: Use `rye`, overhaul settings

### PR DESCRIPTION
**Pull request by khvn26**
This is a chunky one:

## Rye

We rely on Rye for setting up the dev environment and managing dependencies now. By design, Rye adheres to standards as much as it can, so we can use `pyproject.toml` to list dependencies. Rye's UX is a lot like Cargo's of Rust fame, so given that we have Rye installed, we're able to just clone the project and run 

```
rye test
```

to have a working and usable developer environment.

## Entrypoints, project layout, and default `config.json` generation

The project now uses the `src` layout correctly, so a simple `pip install -r requirements.lock` will add the code into `PYTHONPATH`. Additionally, installing the project now adds the following entrypoints:

- `edge-proxy-serve` — runs the Edge Proxy server. `host`, `port` and `reload` Uvicorn configuration entries now settable via Edge Proxy configuration, under the `server` key.
- `edge-proxy-render-config` — writes an example config to stdout and the path specified in `CONFIG_PATH` (`./config.json` by default).

I felt that having `edge-proxy-render-config` will greatly reduce maintenance overhead for keeping the `config.json` up to date.

## `config.json`

The following settings have been renamed, with backwards compatibility retained:

- `api_poll_frequency` to `api_poll_frequency_seconds`.
- `api_poll_timeout` to `api_poll_timeout_seconds`.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,22 +2,18 @@ FROM python:3.12-slim as application
 
 WORKDIR /app
 
-# arm architecture platform builds need postgres drivers installing via apt
-ARG TARGETARCH
-RUN if [ "${TARGETARCH}" != "amd64" ]; then apt-get update && apt-get install -y gcc libpq-dev && rm -rf /var/lib/apt/lists/*; fi;
```
Comment (matthewelwell) [inline Dockerfile:7]: I'm not sure why this was removed?

**Review by matthewelwell — COMMENTED** [review]
One minor question added.

We should perhaps also add a section to the readme for local development? I appreciate there isn't one for the current tooling, but I think it would be wise to add one for rye (especially given that it's a fairly new tool).

File: Dockerfile (hunk 1 above)
Comment (khvn26) [reply Dockerfile:7]: We didn't need those in the first place.

Comment (matthewelwell) [reply Dockerfile:7]: Fair point... I just bothered to actually read the comment.

**Comment by matthewelwell** [comment]
@khvn26 resolved the comment. Do you want to add some detail to the readme (maybe moving over the code in contributing.md added [here](https://github.com/Flagsmith/edge-proxy/pull/103) and fix the conflicts, then we can get this merged? 

I'm keen to do this sooner rather than later, otherwise we're going to be in a constant state of fixing conflicts.

File: CONTRIBUTING.md
Comment (matthewelwell) [inline CONTRIBUTING.md:1]: Ideally we wouldn't remove this file, but just replace it with a copy of the one from the core repo [here](https://github.com/Flagsmith/flagsmith/blob/main/CONTRIBUTING.md)

Comment (khvn26) [reply CONTRIBUTING.md:1]: Done.

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**